AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java

```diff
@@ -85,10 +85,11 @@ protected void processParameters(HasPreferences holder, JMethod listenerMethod,
 		JVar newValueParam = listenerMethod.param(classes().OBJECT, "newValue");
 
 		for (VariableElement variableElement : userParameters) {
-			String type = variableElement.asType().toString();
+			TypeMirror typeMirror = variableElement.asType();
```

Comment:
Maybe this lne should be moved to `castPreferenceIfNecessary()`, since this is present in both handlers.